vxquery-core/src/main/java/org/apache/vxquery/hdfs2/HDFSFunctions.java

```diff
@@ -0,0 +1,463 @@
+     * Read the cluster properties file and locate the HDFS_CONF variable that is the directory path for the
+     * hdfs configuration if the system environment variable HDFS_CONF is not set.
+     * 
+     * @return true if is successfully finds the Hadoop/HDFS home directory
+     */
+    private boolean locateConf() {
+        if (this.conf_path == null) {
+            // load properties file
+            Properties prop = new Properties();
+            String propFilePath = "../vxquery-server/src/main/resources/conf/cluster.properties";
+            try {
+                prop.load(new FileInputStream(propFilePath));
+            } catch (FileNotFoundException e) {
+                propFilePath = "vxquery-server/src/main/resources/conf/cluster.properties";
+                try {
+                    prop.load(new FileInputStream(propFilePath));
+                } catch (FileNotFoundException e1) {
+                } catch (IOException e1) {
+                }
+            } catch (IOException e) {
+                System.err.println(e);
+                return false;
+            }
+            // get the property value for HDFS_CONF
+            this.conf_path = prop.getProperty("HDFS_CONF");
+            if (this.conf_path == null)
+            {
+                this.conf_path = System.getenv("HADOOP_CONF_DIR");
+                return this.conf_path != null;
+            }
+            return this.conf_path != null;
+        }
+        return this.conf_path != null;
+    }
+
+    /**
+     * Upload a file/directory to HDFS.Filepath is the path in the local file system.dir is the destination path.
+     * 
+     * @param filepath
+     * @param dir
+     * @return
+     */
+    public boolean put(String filepath, String dir) {
+        if (this.fs != null) {
+            Path path = new Path(filepath);
+            Path dest = new Path(dir);
+            try {
+                if (fs.exists(dest)) {
+                    fs.delete(dest, true); //recursive delete
+                }
+            } catch (IOException e) {
+                System.err.println(e);
+            }
+            try {
+                fs.copyFromLocalFile(path, dest);
+            } catch (IOException e) {
+                System.err.println(e);
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Get instance of the HDFSfile system if it is configured correctly.
+     * Return null if there is no instance.
+     * 
+     * @return
+     */
+    public FileSystem getFileSystem() {
+        if (locateConf()) {
+            conf.addResource(new Path(this.conf_path + "/core-site.xml"));
+            conf.addResource(new Path(this.conf_path + "/hdfs-site.xml"));
+            try {
+                fs = FileSystem.get(conf);
+                return this.fs;
+            } catch (IOException ex) {
+                System.err.println(ex);
+            }
+        } else {
+            System.err.println("Could not locate hdfs configuarion folder.");
```

Comment:
Should this print to the user or throw an exception. I am concerned we are printing to the user while it should be logged or throw an exception.